src/pudl/analysis/record_linkage/model_helpers.py

```diff
@@ -1,18 +1,134 @@
+
+    tracker_config: ExperimentTrackerConfig
+    run_id: str
+
+    @classmethod
+    def create_experiment_tracker(
+        cls, experiment_config: ExperimentTrackerConfig
+    ) -> "ExperimentTracker":
+        """Create experiment tracker for specified experiment."""
+        run_id = ""
+        if experiment_config.tracking_enabled:
+            mlflow.set_tracking_uri(experiment_config.tracking_uri)
+            with mlflow.start_run(
+                experiment_id=cls.get_or_create_experiment(
+                    experiment_name=experiment_config.experiment_name,
+                    artifact_location=experiment_config.artifact_location,
+                ),
+                tags={"run_context": experiment_config.run_context},
+            ) as run:
+                if experiment_config.log_yaml:
+                    config = flatten_model_config(
+                        get_model_config(experiment_config.experiment_name)
+                    )
+                    mlflow.log_params(config)
+
+                run_id = run.info.run_id
+
+        return cls(tracker_config=experiment_config, run_id=run_id)
+
+    def execute_logging(self, logging_func: Callable):
+        """Perform MLflow logging statement inside ExperimentTracker run.
+
+        Args:
+            logging_func: Callable which should perform an mlflow logging statement
+                inside context manager for run. Passing in callable allows
+                ExperimentTracker to only execute logging if tracking is enabled
+                in configuration.
+        """
+        if self.tracker_config.tracking_enabled:
+            mlflow.set_tracking_uri(self.tracker_config.tracking_uri)
+            with mlflow.start_run(
+                run_id=self.run_id,
+                experiment_id=self.get_or_create_experiment(
+                    self.tracker_config.experiment_name
+                ),
+            ):
+                logging_func()
+
+    @staticmethod
+    def get_or_create_experiment(
+        experiment_name: str, artifact_location: str = ""
+    ) -> str:
+        """Retrieve the ID of an existing MLflow experiment or create a new one if it doesn't exist.
+
+        This function checks if an experiment with the given name exists within MLflow.
+        If it does, the function returns its ID. If not, it creates a new experiment
+        with the provided name and returns its ID.
+
+        Returns:
+            ID of the existing or newly created MLflow experiment.
+        """
+        if experiment := mlflow.get_experiment_by_name(experiment_name):
+            experiment_id = experiment.experiment_id
+        else:
+            experiment_id = mlflow.create_experiment(
+                experiment_name, artifact_location=artifact_location
+            )
+
+        return experiment_id
+
 
-    return model_config
+@op
+def create_experiment_tracker(config: ExperimentTrackerConfig) -> ExperimentTracker:
```

Comment:
This creates an experiment tracker which can be passed around to various `op`s within a `graph`. It will maintain state and restart an `mlflow` everytime it logs something